server/lib/src/entry.rs

```diff
@@ -2401,7 +2403,15 @@ where
             .eclog
             .add_ava_iter(&self.valid.cid, attr, std::iter::once(value.clone()));
         */
-        self.add_ava_int(attr, value)
+        self.add_ava_int(attr, value);
+    }
+
+    pub fn add_ava_if_not_exist(&mut self, attr: &str, value: Value) {
+        // This returns true if the value WAS changed!
```

Comment:
can we add/move this return value docstring to the ada_ava_int function? 🤔 going by the implementation it returns false if it didn't exist before, but was added which is surprising

Reply:
It's the same as the official rust btreeset/hashset, that's why it follows the same pattern. Yes we can move the doc string. :)